src/huggingface_hub/cli/cache.py

```diff
@@ -54,122 +62,93 @@ def _inner(*args, **kwargs):
-        scan_parser.add_argument(
-            "--dir",
-            type=str,
-            default=None,
-            help="cache directory to scan (optional). Default to the default HuggingFace cache.",
-        )
-        scan_parser.add_argument(
-            "-v",
-            "--verbose",
-            action="count",
-            default=0,
-            help="show a more verbose output",
-        )
-        scan_parser.set_defaults(func=CacheCommand, cache_command="scan")
-        # Delete subcommand
-        delete_parser = cache_subparsers.add_parser("delete", help="Delete revisions from the cache directory.")
-        delete_parser.add_argument(
-            "--dir",
-            type=str,
-            default=None,
-            help="cache directory (optional). Default to the default HuggingFace cache.",
-        )
-        delete_parser.add_argument(
-            "--disable-tui",
-            action="store_true",
-            help=(
-                "Disable Terminal User Interface (TUI) mode. Useful if your platform/terminal doesn't support the multiselect menu."
-            ),
-        )
-        delete_parser.add_argument(
-            "--sort",
-            nargs="?",
-            choices=["alphabetical", "lastUpdated", "lastUsed", "size"],
-            help=(
-                "Sort repositories by the specified criteria. Options: "
-                "'alphabetical' (A-Z), "
-                "'lastUpdated' (newest first), "
-                "'lastUsed' (most recent first), "
-                "'size' (largest first)."
-            ),
-        )
-        delete_parser.set_defaults(func=CacheCommand, cache_command="delete")
-
-    def __init__(self, args: Namespace) -> None:
-        self.args = args
-        self.verbosity: int = getattr(args, "verbose", 0)
-        self.cache_dir: Optional[str] = getattr(args, "dir", None)
-        self.disable_tui: bool = getattr(args, "disable_tui", False)
-        self.sort_by: Optional[SortingOption_T] = getattr(args, "sort", None)
-        self.cache_command: Optional[str] = getattr(args, "cache_command", None)
-
-    def run(self):
-        if self.cache_command == "scan":
-            self._run_scan()
-        elif self.cache_command == "delete":
-            self._run_delete()
+@cache_app.command("scan", help="Scan the cache directory")
+def cache_scan(
+    dir: Optional[str] = typer.Option(
+        None,
+        "--dir",
+        help="Cache directory to scan (defaults to Hugging Face cache).",
+    ),
+    verbose: int = typer.Option(
+        0,
+        "-v",
+        "--verbose",
+        count=True,
+        help="Increase verbosity (-v, -vv, -vvv).",
+    ),
+) -> None:
+    _run_scan(cache_dir=dir, verbosity=verbose)
+
+
+def _run_scan(cache_dir: Optional[str], verbosity: int) -> None:
+    try:
+        t0 = time.time()
+        hf_cache_info = scan_cache_dir(cache_dir)
+        t1 = time.time()
+    except CacheNotFound as exc:
+        cache_dir = exc.cache_dir
```

Comment:
I would avoid defining a new internal alias here. A good thing with Typer is that we shouldn't need to define twice some arguments type annotations (and therefore reduce risks of conflicts)

```suggestion
        help="Increase verbosity (-v, -vv, -vvv).",
    ),
) -> None:
    try:
        t0 = time.time()
        hf_cache_info = scan_cache_dir(cache_dir)
        t1 = time.time()
    except CacheNotFound as exc:
        cache_dir = exc.cache_dir
```

Reply:
agree, addressed in  https://github.com/huggingface/huggingface_hub/pull/3365/commits/e5fff141a9f10d5f84c2eeaf15c01eb098a2b257